dashboard/lib/standards_autocomplete.rb

```diff
@@ -0,0 +1,18 @@
+class StandardsAutocomplete < AutocompleteHelper
+  def self.get_search_matches(query, limit, framework_id = nil)
+    limit = format_limit(limit)
+    rows = Standard.limit(limit)
+    rows = rows.where(framework_id: framework_id) if framework_id
+
+    shortcode_rows = rows.where('shortcode like ?', "#{query}%").all
```

Comment:
If it finds shortcode_rows here that match the query does that mean it won't look for descriptions that match the query?

If so I wonder if we should have two input areas instead. One to search shortcodes and one to search descriptions?

Reply:
> If it finds shortcode_rows here that match the query does that mean it won't look for descriptions that match the query?

Correct.

> If so I wonder if we should have two input areas instead. One to search shortcodes and one to search descriptions?

I don't think this is likely to be a problem in practice. The fulltext search only works on words consisting of 3+ consecutive letters (not numbers/symbols). The only framework that has shortcodes starting with 3 letters is CSP, and the only shortcode prefix that is also the prefix of an english word likely to appear in the description is "DAT". so, if you type in "DAT" to the autocomplete, you will see only CSP standards with shortcodes starting with those letters, but as soon as you type "DATA" you will see only standards with the word "data" in the description.

Reply:
Thanks for bringing this up. I hadn't really thought through it until you asked :-)

Reply:
Ok sounds good to me. Thanks for the details!

Reply:
no problem... we can revisit this based on how usable it feels. not sure if this needs to be tracked, but another quirk is that the backend does a natural "boolean mode" search, where a phrase like "computer security" can match a standard with a description where those words are not consecutive: https://github.com/code-dot-org/code-dot-org/blob/a13dbde5c38ac664228148ecaac12891509a711e/dashboard/test/lib/standards_autocomplete_test.rb#L38-L42 https://github.com/code-dot-org/code-dot-org/blob/ce94aad06fedd6acfa202264db271f576fd4cafe/dashboard/config/standards/csta_standards.csv#L118 however, the autocomplete searchbox then filters those out because it does not have the exact string "computer security" in it. 🤷  CC @bethanyaconnor